File: DsAlgo/CanhCuathanKy.cpp
#include <bits/stdc++.h>
using namespace std;
typedef pair<string,int> psi;
int main()
{
	freopen("input.txt","r",stdin);
	int test;
	cin>>test;
	while(test--)
	{
		queue<psi> Q;
		string res[]={"dangdungcntt", "tienquanutc", "quang123", "maianh", "nguyenminhduc2820"};
		for(auto x:res) Q.push({x,1});
		int n;
		cin>>n;
		while(n>Q.front().second)
		{
			n-=Q.front().second;
			Q.push({Q.front().first,Q.front().second*2});
			Q.pop();
		}
		cout<<Q.front().first<<"\n";
	}
}

File: DsAlgo/BuffToTest/CanhCuaThanKy.py
from queue import Queue
if __name__ == '__main__':
    a =['dangdungcntt','tienquanutc','quang123','maianh','nguyenminhduc2820']
    for _ in range(int(input())):
        n = int(input())
        Q = Queue()
        for i in [0,1,2,3,4]: Q.put((i,1))
        while n>Q.queue[0][1]:
            x = Q.get()
            n-=x[1]
            Q.put((x[0],x[1]*2))
        print(a[Q.get()[0]])

